pkg/util/metrics_helper.go

```diff
@@ -162,6 +162,22 @@ func (d MetricFamiliesPerUser) SendSumOfGaugesWithLabels(out chan<- prometheus.M
 	d.sumOfSingleValuesWithLabels(gauge, gaugeValue, labelNames).WriteToMetricChannel(out, desc, prometheus.GaugeValue)
 }
 
+// SendSumOfGaugesPerUserWithLabels provides metrics with the provided label names on a per-user basis. This function assumes that `user` is the
+// first label on the provided metric Desc
+func (d MetricFamiliesPerUser) SendSumOfGaugesPerUserWithLabels(out chan<- prometheus.Metric, desc *prometheus.Desc, metric string, labelNames ...string) {
```

Comment:
@pstibrany Could you review this function, please?

Reply:
To fit the rest of the code, here is my suggestion:

```suggestion
func (d MetricFamiliesPerUser) SendSumOfGaugesPerUserWithLabels(out chan<- prometheus.Metric, desc *prometheus.Desc, metric string, labelNames ...string) {
	for user, userMetrics := range d {
		result := singleValueWithLabelsMap{}
		userMetrics.sumOfSingleValuesWithLabels(metric, labelNames, gaugeValue, result.aggregateFn)

		for key, mlv := range result {
			mlv.LabelValues = append([]string{user}, mlv.LabelValues...)
			result[key] = mlv
		}

		result.WriteToMetricChannel(out, desc, prometheus.GaugeValue)
	}
}
```

where the inner `for` for adding user label could even be moved to `singleValueWithLabelsMap` type as a new method.

Reply:
I will use your suggested snippet and move that `for` loop to a method called `appendUserLabelValue` for the `singleValueWithLabelsMap` struct